charts/sn-platform/templates/vault/vault-update.yaml

```diff
@@ -0,0 +1,91 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+{{- if .Values.components.vault }}
+{{- if .Values.vault.oidcToken.update }}
```

Comment:
Genuine question, do you need this?

The annotation on the hook means the job will only happen on upgrades anyway. Is there any harm to not have this guard?

Reply:
How about removing this switch and use the option `--no-hooks` instead when doing `helm install/upgrade`?

Reply:
@fantapsody yes, I think it's a good idea, but we hope it don't exec when first install, only exec when you want to update, but as you know Users get used to the default behavior without `--no-hooks` @tuteng how do you think?

Reply:
@fantapsody yes, good idea, fixed